middlewares/validators/validator.helper.js

```diff
@@ -386,84 +424,133 @@ function searchValidator (fieldLocation, fieldname) {
+        });
 }
 
-function searchSortValidator (fieldLocation, fieldName){
+function searchSortValidator(fieldLocation, fieldName) {
     const searchSort = setProperValidationChainBuilder(fieldLocation, fieldName, "Invalid sort criteria")
-    return searchSort.optional({checkFalsy: true})
+    return searchSort.optional({
+            checkFalsy: true
+        })
         .custom((value) => {
             let modelString = param("model", "Corresponding model not found");
             let model;
-            if(modelString.equals("hacker")){
+            if (modelString.equals("hacker")) {
                 model = Models.Hacker;
-            }
-            else{
+            } else {
                 return false;
             }
-            if(!!model.searchableField(value)){
+            if (!!model.searchableField(value)) {
                 let sortOrder = param("sort", "Sorting order not found");
-                if(!sortOrder.equals("asc") || !sortOrder.equals("desc")){
+                if (!sortOrder.equals("asc") || !sortOrder.equals("desc")) {
                     return false;
                 }
-            }
-            else{
+            } else {
                 return false;
             }
             return true;
         })
 }
 
+/**
+ * Validates that field must be a valid date according to javascript Date.parse
+ * @param {"query" | "body" | "header" | "param"} fieldLocation the location where the field should be found 
+ * @param {string} fieldname name of the field that needs to be validated.
+ * @param {boolean} optional whether the field is optional or not.
+ */
+function dateValidator(fieldLocation, fieldname, optional = true) {
+    const date = setProperValidationChainBuilder(fieldLocation, fieldname, "invalid date");
+    if (optional) {
+        return date.optional({
+            checkFalsy: true
+        }).custom(value => {
+            return !isNaN(Date.parse(value));
+        }).withMessage({
+            message: "Date is not valid.",
+            isValid: date
+        });
+    } else {
+        return date.exists().withMessage("Date field must be specified").custom(value => {
+            return !isNaN(Date.parse(value));
+        }).withMessage({
+            message: "Date is not valid.",
+            isValid: date
+        });
+    }
+}
+
+/**
+ * Validates that field must be a valid phone number, having numbers and having a minimum length of 8
+ * @param {"query" | "body" | "header" | "param"} fieldLocation the location where the field should be found 
+ * @param {string} fieldname name of the field that needs to be validated.
+ * @param {boolean} optional whether the field is optional or not.
+ */
+function phoneNumberValidator(fieldLocation, fieldname, optional = true) {
+    const number = setProperValidationChainBuilder(fieldLocation, fieldname, "invalid phone number");
+    if (optional) {
+        return number.optional({
+            checkFalsy: true
+        }).isLength({
+            min: 8
```

Comment:
Can we do a more proper validation for phone number, such as a regexp? I’m sure there exists some online that we can use